scripts/integration-test.sh

```diff
@@ -1,6 +1,531 @@
+    > crontabber_app.log 2>&1
+check_for_logged_fatal_errors $? crontabber_app
+echo " Done."
+
+
+#------------------------------------------------------------------------------
+echo "INFO: ensure no processes already running..."
+cleanup
+echo " Done."
+
+#******************************************************************************
+# Here's where we actually start testing
+# Iterate through some combinations of collector/crashmover/processor/middleware/setups
+# These setups are defined in functions with their names list in the for loop:
+for an_app_set in start_2015_socorro_apps start_standard_socorro_apps start_minimal_socorro_apps
+do
+  # start the apps by invoking their function
+  $an_app_set
+
+  # wait for collector to startup
+  retry collector "running standalone at 0.0.0.0:8882"
+
+  #----------------------------------------------------------------------------
+  # BREAKPAD submission test
+  echo 'INFO: submitting breakpad test crash...'
+  # submit test crash
+  socorro submitter \
+      -u http://0.0.0.0:8882/submit \
+      -s testcrash/raw/ \
+      -n 1 \
+      > submitter.log 2>&1
+  check_for_logged_fatal_errors $? submitter
+
+
+  CRASHID=$(grep 'CrashID' submitter.log | awk -FCrashID=bp- '{print $2}')
+  echo found $CRASHID
+  if [ -z "$CRASHID" ]
+  then
+    echo "ERROR: CRASHID missing from submitter.log"
+    echo "***** BEGIN collector log *****"
+    cat collector.log
+    echo "***** END collector log *****"
+    echo "***** BEGIN submitter log *****"
+    cat submitter.log
+    echo "***** END submitter log *****"
+    fatal 1 "no crash ID found in submitter log"
+  fi
+  echo "INFO: collector received crash ID: $CRASHID"
+  echo " Done."
+
+  #----------------------------------------------------------------------------
+  # make sure crashes are picked up, and no errors are logged
+  retry 'collector' "$CRASHID"
+  retry 'processor' "saved - $CRASHID"
+
+  #----------------------------------------------------------------------------
+  # check that mware has raw crash using curl to hit the HTTP endpoint
+  curl -s -D middleware_headers.log "http://localhost:8883/crash_data/?datatype=meta&uuid=$CRASHID" > /dev/null
+  err=$?
+  echo "  looking for errors in hitting the middleware for $CRASHID"
+  check_for_logged_fatal_errors $err middleware
+
+  echo "  looking for "200 OK" in hitting the middleware for $CRASHID"
+  grep '200 OK' middleware_headers.log > /dev/null
+  fatal $? "middleware test failed, no raw data for crash ID $CRASHID"
+
+  echo "  looking for processed crash through middleware for $CRASHID"
+  function find_crash_in_middleware() {
+    curl -s "http://localhost:8883/crash_data/?datatype=processed&uuid=$CRASHID" | grep date_processed
+    echo "http://localhost:8883/crash_data/?datatype=processed&uuid=$CRASHID"
+    return $?
+  }
+  retry_command middleware find_crash_in_middleware
+
+  # check that mware logs the request for the crash, and logs no errors
+  retry 'middleware' "/crash_data"
```

Comment:
This all we'll have to remember to re-evaluate when we do away with the middleware.